### PR TITLE
Update the dynamic volume limit in GCE PD

### PR DESCRIPTION
**Pull request by jingxu97**
Currently GCE PD support 128 maximum disks attached to a node for all
machines types except shared-core. This PR updates the limit number to
date.

Change-Id: Id9dfdbd24763b6b4138935842c246b1803838b78

```release-note
Change the maximum disks attached per node for GCE PDs to 128 for all machines types except shared-core which will be 16.
```

### CODE REVIEW

**Comment by saad-ali** [comment]
Assigning patch release managers:
/assign @aleksandra-malinowska
/assign @tpepper

**Review by saad-ali — APPROVED** [review]

**Comment by tpepper** [comment]
/hold
pending description of what bug is solved by this.

**Comment by saad-ali** [comment]
@tpepper The GCE platform changed the number of disks that are supported per node. These PRs are updating those limits on the k8s side. The change is on the edge between a bug and a feature. But since we've gotten customer complaints about it (it appears to customers like a bug), we'd like to cherry pick it.

**Comment by tpepper** [comment]
Since this is small, provider specific, and SIG approved, will merge.

**Pull request merged**
